lib/papertrail/connection.rb

```diff
@@ -177,55 +173,47 @@ def each_event(query_term = nil, options = {}, &block)
       end
 
       # Remove all of the options that shouldn't be in each query
-      options  = options.dup
+      options  = { :tail => false }.merge(options)
       min_id   = options.delete(:min_id)
       max_id   = options.delete(:max_id)
       min_time = options.delete(:min_time)
       max_time = options.delete(:max_time)
 
       # Figure out where to start querying
       if min_id
-        search_results = query(query_term, options.merge(:min_id => min_id, :tail => false)).search
+        search = search_query(query_term, options.merge(:min_id => min_id))
       elsif min_time
-        search_results = query(query_term, options.merge(:min_time => min_time.to_i, :tail => false)).search
+        search = search_query(query_term, options.merge(:min_time => min_time.to_i))
       else
         raise ArgumentError, "Either :min_id or :min_time must be specified"
       end
 
       # Start processing events
       loop do
+        search_results = search.results_page
         search_results.events.each do |event|
           # If we've found an event beyond what we were looking for, we're done
-          if max_time && event.received_at > max_time
-            break
-          end
-
-          if max_id && event.id > max_id
-            break
-          end
+          break if max_time && event.received_at > max_time
+          break if max_id && event.id > max_id
 
           block.call(event)
         end
 
         # If we've found the end of what we're looking for, we're done
-        if max_time && search_results.max_time_at > max_time
-          break
-        end
-
-        if max_id && search_results.max_id > max_id
-          break
-        end
+        break if max_time && search_results.max_time_at > max_time
+        break if max_id && search_results.max_id > max_id
```

Comment:
I can imagine a time that the API returns a later `max_time_at` than any of the events - but is it really possible that there will be a time when the API returns an event later than the `max_time_at`? If not, we can remove the individual event iteration and just rely on the `max_time_at` and `max_id` parameters?